run.go

```diff
@@ -434,76 +350,61 @@ func (d *InMAP) checkOutputNames(o map[string]string) error {
 // 'coxHazard(Relative Risk, Population, Mortality Rate)' which calculates a
 // deaths estimate based on the relative risk associated with PM 2.5 changes
 // and the baseline number of deaths.
-func (d *InMAP) Results(allLayers bool, checkNames bool, outputVariables map[string]string) (map[string][]float64, error) {
-
-	functions := map[string]govaluate.ExpressionFunction{
-		"exp": func(arg ...interface{}) (interface{}, error) {
-			if len(arg) != 1 {
-				return nil, fmt.Errorf("inmap: got %d arguments for function 'exp', but needs 1", len(arg))
-			}
-			return (float64)(math.Exp(arg[0].(float64))), nil
-		},
-		"loglogRR": func(arg ...interface{}) (interface{}, error) {
-			if len(arg) != 1 {
-				return nil, fmt.Errorf("inmap: got %d arguments for function 'exp', but needs 1", len(arg))
-			}
-			return (float64)(aqhealth.RRpm25Linear(arg[0].(float64))), nil
-		},
-		"coxHazard": func(args ...interface{}) (interface{}, error) {
-			if len(args) != 3 {
-				return nil, fmt.Errorf("inmap: got %d arguments for function 'exp', but needs 3", len(args))
-			}
-			return (float64)((args[0].(float64) - 1) * args[1].(float64) * args[2].(float64) / 100000), nil
-		},
-	}
-
-	outputVariables, getVariables, err := checkForDerivitives(outputVariables, functions)
-	if err != nil {
-		return nil, err
-	}
-
-	if err := d.checkGetNames(getVariables...); err != nil {
-		return nil, err
-	}
-
-	if checkNames {
-		if err := d.checkOutputNames(outputVariables); err != nil {
-			return nil, err
-		}
-	}
+func (d *InMAP) Results(o *Outputter) (map[string][]float64, error) {
 
 	// Prepare output data
-	getConc := make(map[string][]float64)
-	concByRow := make(map[string]interface{})
-	outputConc := make(map[string][]float64)
+	getVal := make(map[string][]float64)
+	preproccessVals := make(map[string]interface{})
+	valByRow := make(map[string]interface{})
+	outputVal := make(map[string][]float64)
 	var nCells int
 
-	for _, name := range getVariables {
-		if allLayers {
+	for _, name := range o.GetVariables {
+		if o.AllLayers {
 			data := d.toArray(name, -1)
-			getConc[name] = data
+			getVal[name] = data
+			preproccessVals[name] = data
 			nCells = len(data)
 		} else {
 			data := d.toArray(name, 0)
-			getConc[name] = data
+			getVal[name] = data
+			preproccessVals[name] = data
 			nCells = len(data)
 		}
 	}
-	for k, v := range outputVariables {
-		expression, err := govaluate.NewEvaluableExpressionWithFunctions(v, functions)
+
+	for k, v := range o.OutputVariables {
```

Comment:
Would you be willing to add some comments describing what is going on here? It looks like you added a test to make sure it works, but I don't understand it at all at this point :)